code/game/objects/items/nemesis_items.dm

```diff
@@ -0,0 +1,458 @@
+
+/obj/item/nemesis_mine
+	name = "energy mine"
+	desc = "A high-tech trap, designed to electrocute any intruders that step on it."
+	icon = 'icons/obj/nemesis.dmi'
+	icon_state = "nemesis_trap_inactive"
+	w_class = WEIGHT_CLASS_SMALL
+
+	var/active = FALSE
+	var/overloaded = FALSE
+	var/obj/item/clothing/glasses/hud/security/sunglasses/nemesis/glasses
+
+/obj/item/nemesis_mine/Crossed(mob/living/AM)
+	. = ..()
+
+	if(!istype(AM) || !isturf(loc) || !active)
+		return
+
+	if(AM.movement_type & FLYING)
+		return
+
+	trigger(AM)
+
+/obj/item/nemesis_mine/attack_self(mob/user)
+	if(!ishuman(user))
+		return ..()
+
+	var/mob/living/carbon/human/owner = user
+	var/obj/item/clothing/gloves/rapid/nemesis/gloves = owner.get_item_by_slot(ITEM_SLOT_GLOVES)
+	if(!gloves || !istype(gloves))
+		return ..()
+
+	if(overloaded)
+		gloves.gain_charge(round(NEMESIS_CHARGE_PER_MINE * 0.5)) //Small refund
+	else
+		if(gloves.charge < NEMESIS_CHARGE_PER_MINE)
+			to_chat(user, "<span class='warning'>Your A.R.E.S. suit does not posess enough charge to overload [src]! ([gloves.charge]/[NEMESIS_CHARGE_PER_MINE])</span>")
+			return ..()
+
+		gloves.lose_charge(NEMESIS_CHARGE_PER_MINE)
+
+	overloaded = !overloaded
+	to_chat(user, "<span class='notice'>You [overloaded ? "overload [src]'s detection system" : "return [src]'s circutry back to the normal state"].</span>")
+	icon_state = "[overloaded ? "nemesis_trap" : "nemesis_trap_inactive"]"
+
+/obj/item/nemesis_mine/attack_hand(mob/living/user, list/modifiers)
+	if(!active)
+		return ..()
+
+	to_chat(user, "<span class='warning'>You attempt to disarm [src]...</span>")
+	var/disarm_time = 15 SECONDS
+	if(ishuman(user))
+		var/mob/living/carbon/human/disarmer = user
+		if(istype(disarmer.get_item_by_slot(ITEM_SLOT_GLOVES), /obj/item/clothing/gloves/rapid/nemesis))
+			disarm_time = 1 SECONDS
+
+	if(!do_after(user, disarm_time, src))
+		return
+	active = FALSE
+	icon_state = initial(icon_state)
+	update_icon()
+	anchored = FALSE
+	to_chat(user, "<span class='notice'>You successfully disarm [src].</span>")
+
+/obj/item/nemesis_mine/attackby(obj/O, mob/user, params)
+	if(!active)
+		return ..()
+
+	if(!istype(O, /obj/item/screwdriver))
+		return ..()
+
+	to_chat(user, "<span class='warning'>You attempt to disarm [src]...</span>")
+	var/disarm_time = 3 SECONDS
+	if(ishuman(user))
+		var/mob/living/carbon/human/disarmer = user
+		if(istype(disarmer.get_item_by_slot(ITEM_SLOT_GLOVES), /obj/item/clothing/gloves/rapid/nemesis))
+			disarm_time = 1 SECONDS
+
+	if(!do_after(user, disarm_time, src))
+		return
+
+	active = FALSE
+	icon_state = initial(icon_state)
+	update_icon()
+	anchored = FALSE
+	to_chat(user, "<span class='notice'>You successfully disarm [src].</span>")
+
+
+/obj/item/nemesis_mine/proc/trigger(mob/living/victim)
+	var/turf/cur_loc = get_turf(src)
+	new /obj/effect/temp_visual/nemesis_circle(cur_loc)
+
+	if(!victim)
+		victim = locate() in cur_loc
+
+	if(victim)
+		victim.apply_damage(75, STAMINA)
```

Comment:
can you put two of these on one tile to dab on anyone who walks onto that tile?

Reply:
Yeeeah, I should probably disallow players to make such shit